trace_viewer/core/tracks/memory_dump_track.html

```diff
@@ -0,0 +1,200 @@
+        ctx.beginPath();
+        ctx.arc(xView, halfHeight, dumpRadiusView + 0.5, 0, twoPi);
+        ctx.fill();
+        if (memoryDump.selected) {
+          ctx.lineWidth = 3;
+          ctx.strokeStyle = 'rgb(100,100,0)';
+          ctx.stroke();
+
+          ctx.beginPath();
+          ctx.arc(xView, halfHeight, dumpRadiusView, 0, twoPi);
+          ctx.lineWidth = 1.5;
+          ctx.strokeStyle = 'rgb(255,255,0)';
+          ctx.stroke();
+        } else {
+          ctx.lineWidth = 1;
+          ctx.strokeStyle = 'rgb(0,0,0)';
+          ctx.stroke();
+        }
+
+        ctx.fillStyle = 'rgb(255, 255, 255)';
+        ctx.font = '400 9px Arial';
+        ctx.textBaseline = 'middle';
+        ctx.textAlign = 'center';
+        ctx.fillText('M', xView, halfHeight);
+      }
+      ctx.lineWidth = 1;
+
+      // For performance reasons we only check the SelectionState of the first
+      // memory dump. If it's DIMMED we assume that all are DIMMED.
+      // TODO(petrcermak): Allow partial highlight.
+      var selectionState = SelectionState.NONE;
+      if (memoryDumps.length &&
+          memoryDumps[0].selectionState === SelectionState.DIMMED) {
+        selectionState = SelectionState.DIMMED;
+      }
+
+      // Dim the track when there is an active highlight.
+      if (selectionState === SelectionState.DIMMED) {
+        var width = bounds.width * pixelRatio;
+        ctx.fillStyle = 'rgba(255,255,255,0.5)';
+        ctx.fillRect(0, 0, width, height);
+      }
+    },
+
+    addEventsToTrackMap: function(eventToTrackMap) {
+      this.memoryDumps.forEach(function(obj) {
+        eventToTrackMap.addEvent(obj, this);
+      }, this);
+    },
+
+    addIntersectingItemsInRangeToSelectionInWorldSpace: function(
+        loWX, hiWX, viewPixWidthWorld, selection) {
+      var memoryDumpRadiusWorld = viewPixWidthWorld * this.dumpRadiusView;
+      tv.b.iterateOverIntersectingIntervals(
+          this.memoryDumps,
+          function(x) { return x.start - memoryDumpRadiusWorld; },
+          function(x) { return 2 * memoryDumpRadiusWorld; },
+          loWX, hiWX,
+          selection.push.bind(selection));
+    },
+
+    /**
+     * Add the item to the left or right of the provided event, if any, to the
+     * selection.
+     * @param {event} The current event item.
+     * @param {Number} offset Number of slices away from the event to look.
+     * @param {Selection} selection The selection to add an event to,
+     * if found.
+     * @return {boolean} Whether an event was found.
+     * @private
+     */
+    addItemNearToProvidedEventToSelection: function(event, offset, selection) {
+      var events = this.memoryDumps;
+      var index = events.indexOf(event);
+      var newIndex = index + offset;
+      if (newIndex >= 0 && newIndex < events.length) {
+        selection.push(events[newIndex]);
+        return true;
+      }
+      return false;
+    },
+
+    addAllObjectsMatchingFilterToSelection: function(filter, selection) {
```

Comment:
I don't see any reason why that should not be possible. In fact, I believe that is part of our plan :-)